### PR TITLE
att4425: new skeleton (AT&T 4425 / Teletype 56D text terminal)

### PR DESCRIPTION
**Pull request by shattered**

### CODE REVIEW

File: scripts/target/mame/mess.lua [hunk 1]
```diff
@@ -3006,6 +3006,11 @@ files {
 	MAME_DIR .. "src/mame/includes/tmc2000e.h",
 }
 
+createMESSProjects(_target, _subtarget, "teletype")
+files {
+	MAME_DIR .. "src/mame/drivers/att4425.cpp",
+}
+
```
Comment (cuavas) [inline scripts/target/mame/mess.lua]: Can you please put this in the skeleton project for now?  We already have a plethora of single-driver projects.

File: src/mame/drivers/att4425.cpp [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+		, m_p_chargen(*this, "chargen")
+		, m_screen(*this, "screen")
+		{ }
+
+	DECLARE_PALETTE_INIT(att4425);
+
+	DECLARE_WRITE_LINE_MEMBER(write_line_clock);
+	DECLARE_WRITE_LINE_MEMBER(write_keyboard_clock);
+
+	virtual void machine_start() override;
```
Comment (cuavas) [inline src/mame/drivers/att4425.cpp]: Can you please make overrides of protected member function protected, and make data members protected or private?  We're trying to encourage good habits.

File: src/mame/drivers/att4425.cpp [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+{
+public:
+	att4425_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, Z80_TAG)
+		, m_i8251(*this, I8251_TAG)
+		, m_sio(*this, Z80SIO_TAG)
+		, m_p_videoram(*this, "videoram")
+		, m_p_chargen(*this, "chargen")
+		, m_screen(*this, "screen")
```
Comment (cuavas) [inline src/mame/drivers/att4425.cpp]: If you're defining a macro for the screen tag above, you should probably use it when constructing the device finder here.

File: src/mame/drivers/att4425.cpp [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+static INPUT_PORTS_START( att4425 )
+INPUT_PORTS_END
+
+/* Video */
+
+PALETTE_INIT_MEMBER(att4425_state, att4425)
+{
+	palette.set_pen_color(0, 0, 0, 0);   /* Black */
+	palette.set_pen_color(1, 0, 255, 0); /* Normal */
+	palette.set_pen_color(2, 0, 192, 0); /* Dim */
```
Comment (cuavas) [inline src/mame/drivers/att4425.cpp]: We already have a setup for doing monochrome with hilight, see e.g. osborne1.cpp for a example.  The phosphor colour is an attribute of the screen, and a monochrome palette is used.

```
    MCFG_SCREEN_ADD_MONOCHROME("screen", RASTER, rgb_t::green())
    MCFG_SCREEN_UPDATE_DRIVER(osborne1_state, screen_update)
    MCFG_SCREEN_RAW_PARAMS( MAIN_CLOCK, 1024, 0, 104*8, 260, 0, 24*10 )
    MCFG_SCREEN_PALETTE("palette")
    MCFG_GFXDECODE_ADD("gfxdecode", "palette", osborne1)
    MCFG_PALETTE_ADD_MONOCHROME_HIGHLIGHT("palette")
```

**Comment by shattered** [comment]
New PR with requested changes: https://github.com/mamedev/mame/pull/2584

**Pull request closed without merging**